viewing_party/party.py

```diff
@@ -1,23 +1,137 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if (not title) or (not genre) or (not rating):
+        return None
+      
+    if not all((isinstance(title, str), isinstance(genre, str),
+               isinstance(rating, float))):
+        return None
+
+    new_movie = {}
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+
+    return new_movie
+
+
+def add_to_watched(user_data, movie):
+    # Handles if create_movie() returns None for movie
+    if movie:
+        user_data["watched"].append(movie)
+       
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    if movie:
+        user_data["watchlist"].append(movie)
+   
+    return user_data
+
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            move_movie = movie
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(move_movie)
+
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
 
+def get_watched_avg_rating(user_data):
+    sum = 0

```

Comment:
Consider moving any variables below your guard clauses. That way if the conditionals are executed, we haven't made variables in memory that will then be immediately tossed out.